lib/fireblocks/api/request.rb

```diff
@@ -41,21 +41,24 @@ def put(body)
       valid_response!(send_request(req), request: req)
     end
 
-    def post(body)
+    def post(body, headers)
       req = Net::HTTP::Post.new(uri)
-      request_headers(body).each { |rk, rv| req[rk] = rv }
+      request_headers(body, headers).each { |rk, rv| req[rk] = rv }
+      req['Idempotency-Key'] = idempotency_key if idempotency_key
+      @idempotency_key = idempotency_key
       req.body = body.to_json
 
       valid_response!(send_request(req), request: req)
     end
 
     private
 
-    def request_headers(body)
+    def request_headers(body, headers = {})
```

Comment:
Keyword args would be better here 

```suggestion
    def request_headers(body, headers: {})
```

Reply:
@snogrammer I reverted back to default args bc the `idempotency-key` wasn't being passed through headers with the keyword args